src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2083,6 +2083,14 @@ PeerImp::getLedger (std::shared_ptr<protocol::TMGetLedger> const& m)
     p_journal_.trace <<
         "GetLeder: " << logMe;
 
+    int depth = -1;
+    if (packet.has_querydepth())
+        depth = packet.querydepth();
```

Comment:
If the query depth is negative it means its up to the discretion of the other end to decide the depth.